### PR TITLE
Drop Node 16 support

### PR DESCRIPTION
**Pull request by GytisCepk**
### Node 16 will reach End-of-Life later this year ([Node.js release schedule](https://github.com/nodejs/release#release-schedule)).

PR Changes:

- Change ranges to support only Node 18.
- Remove Node 16 runs from integration and regressions pipelines.
- Update documentation.
- Update `object-storage` packages versions to `^1.5.0`.

### CODE REVIEW

**Review by MichaelBelousov — APPROVED** [review]
transformer changes are fine

**Review by aruniverse — APPROVED** [review]

**Comment by saskliutas** [comment]
`@itwin/core-frontend` dependency [`"@itwin/cloud-agnostic-core": "~1.4.0"`](https://github.com/iTwin/itwinjs-core/blob/742071bf231c5ecd029c14d42148b7b91e1cab4a/core/frontend/package.json#L97) does not support Node 18. Shouldn't it be updated before this?

**Review by ben-polinsky — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Review by paulius-valiunas — APPROVED** [review]

File: common/config/azure-pipelines/jobs/regression-testing.yaml [hunk 1]
```diff
@@ -43,22 +43,6 @@ jobs:
           platform: Darwin
           name: $(mac_pool)
           node_version: 18.12.0
-
```
Comment (pmconne) [inline common/config/azure-pipelines/jobs/regression-testing.yaml:46]: Comments at top of file are out of date, if it matters.

**Review by pmconne — APPROVED** [review]

File: core/backend/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "typings": "lib/cjs/core-backend",
   "license": "MIT",
   "engines": {
-    "node": ">=16.13.0 < 19.0"
+    "node": "^18.0.0"
```
Comment (aruniverse) [inline core/backend/package.json:9]: maybe we should actually set the engine to be ^18.12.0; which is what we're saying is the minimum compatible version in some of the documentation, and the testing pipelines

File: core/backend/package.json [hunk 3]
```diff
@@ -92,9 +92,9 @@
   "dependencies": {
```
Comment (aruniverse) [inline core/backend/package.json:92]: could you also try updating the types/node dev dep to use ^ for all pkgs?

**Review by aruniverse — APPROVED** [review]

File: core/backend/package.json (hunk 2 above)
Comment (GytisCepk) [reply core/backend/package.json:9]: Used [this discussion](https://github.com/iTwin/itwinjs-core/pull/4295#discussion_r1005113967) as a reference to what range this should be set. In docs, used `^18.12.0` to encourage using LTS versions. 

`^18.12.0` would make thinks more consistent.

Comment (aruniverse) [reply core/backend/package.json:9]: thx for bringing that back up, i guess based on calebs comment we should leave as is then

File: core/backend/package.json (hunk 3 above)
Comment (GytisCepk) [reply core/backend/package.json:92]: Updated

**Pull request merged**